R/pmap.R

```diff
@@ -101,11 +101,14 @@ pmap_chr <- function(.l, .f, ..., .progress = FALSE) {
 
 pmap_ <- function(.type, .l, .f, ..., .progress = FALSE, ..error_call = caller_env()) {
   .f <- as_mapper(.f, ...)
-  if (is.data.frame(.l)) {
-    .l <- as.list(.l)
-  }
-  i <- 0L
 
+  .l <- vctrs_list_compat(.l)
+  vec_check_list(.l, call = ..error_call)
+
+  .l <- map(.l, vctrs_list_compat)
+  .l <- vec_recycle_common(!!!.l, .arg = ".l", .call = ..error_call)
```

Comment:
At this point should `list_check_all_vectors()` also be used? It seems like there were some C checks to ensure that each element is a vector

Reply:
Hmm, I guess it errors on `NULL` list elements and it seems like we don't want that?

Reply:
Oh actually `vec_recycle_common()` will take care of this for us, and handles `NULL` right too